### PR TITLE
Rework log file machinery to allow log-file logging to be repeatedly …

### LINKED ISSUE #1456: TestStatusLocalLogs fails with multiple CPUs
repro: `make CPUS=2,3,4 PKG=./server TESTS=TestStatusLocalLogs`

Subtask of #1406.

### PR DESCRIPTION
**Pull request by spencerkimball**
…reset.

Fixes #1456.

### CODE REVIEW

File: util/log/file.go [hunk 1]
```diff
@@ -165,37 +155,33 @@ func parseLogFilename(filename string) (FileDetails, error) {
...
-// contains severity ("INFO", "FATAL", etc.) and t.  If the file is created
+// contains severity ("INFO", "FATAL", etc.) and t. If the file is created
 // successfully, create also attempts to update the symlink for that tag, ignoring
 // errors.
 func create(severity Severity, t time.Time) (f *os.File, filename string, err error) {
-	onceLogDirs.Do(createLogDirs)
-	if len(logDirs) == 0 {
-		return nil, "", errors.New("log: no log dirs")
+	if len(*logDir) == 0 {
+		return nil, "", errors.New("log: log directory empty")
```
Comment (tamird) [inline util/log/file.go]: s/empty/unset/

current wording implies that the log directory doesn't contain files

File: util/log/file.go [hunk 2]
```diff
@@ -165,37 +155,33 @@ func parseLogFilename(filename string) (FileDetails, error) {
...
-		// Append is almost always more efficient than O_RDRW on most modern file systems.
-		f, err = os.OpenFile(fname, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0664)
-		if err != nil {
-			return nil, "", fmt.Errorf("log: cannot create log: %v", err)
-		}
+	// Open the file os.O_APPEND|os.O_CREATE rather than use os.Create.
+	// Append is almost always more efficient than O_RDRW on most modern file systems.
+	f, err = os.OpenFile(fname, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0664)
+	if err != nil {
+		return nil, "", fmt.Errorf("log: cannot create log: %v", err)
```
Comment (tamird) [inline util/log/file.go]: looks redundant with what's below

File: util/log/log.go [hunk 3]
```diff
@@ -45,6 +50,10 @@ func EnableLogFileOutput(dir string) {
 
 // DisableLogFileOutput turns off logging. For unittesting only.
 func DisableLogFileOutput() {
+	if err := logging.removeFiles(); err != nil {
+		fmt.Fprintf(os.Stderr, "unable to remove log files: %s", err)
```
Comment (tamird) [inline util/log/log.go]: i think you want to use `logging.exit` here

**Comment by tamird** [comment]
LGTM after comments

File: util/log/file.go [hunk 4]
```diff
@@ -165,37 +155,33 @@ func parseLogFilename(filename string) (FileDetails, error) {
...
-		return nil, "", errors.New("log: no log dirs")
+	if len(*logDir) == 0 {
+		return nil, "", errors.New("log: log directory empty")
 	}
 	name, link := logName(severity, t)
 	var lastErr error
-	for _, dir := range logDirs {
-		fname := filepath.Join(dir, name)
+	fname := filepath.Join(*logDir, name)
+	fmt.Printf("trying to create log file %s\n", fname)
```
Comment (bdarnell) [inline util/log/file.go]: Remove this.

File: util/log/file.go (hunk 2 above)
Comment (spencerkimball) [inline util/log/file.go]: Removed.

File: util/log/file.go (hunk 1 above)
Comment (spencerkimball) [inline util/log/file.go]: Changed.

**Comment by bdarnell** [comment]
LGTM

File: util/log/log.go (hunk 3 above)
Comment (spencerkimball) [inline util/log/log.go]: Changed.

File: util/log/file.go (hunk 4 above)
Comment (spencerkimball) [inline util/log/file.go]: Removed.

**Pull request merged**